src/tensors/abstracttensor.jl

```diff
@@ -205,29 +216,90 @@ See also [`blocks`](@ref), [`blocksectors`](@ref), [`blockdim`](@ref) and [`hasb
 """ block
 
 """
-    hasblock(t::AbstractTensorMap, c::Sector) -> Bool
+    blocksectors(t::AbstractTensorMap)
 
-Verify whether a tensor has a block corresponding to a coupled sector `c`.
+Return an iterator over all coupled sectors of a tensor.
 """
-hasblock
+blocksectors(t::AbstractTensorMap) = keys(tensorstructure(t).blockstructure)
 
-@doc """
-    blocksectors(t::AbstractTensorMap)
+"""
+    hasblock(t::AbstractTensorMap, c::Sector) -> Bool
 
-Return an iterator over all coupled sectors of a tensor.
-""" blocksectors(::AbstractTensorMap)
+Verify whether a tensor has a block corresponding to a coupled sector `c`.
+"""
+hasblock(t::AbstractTensorMap, c::Sector) = c ∈ blocksectors(t)
 
-@doc """
-    blockdim(t::AbstractTensorMap, c::Sector) -> Base.Dims
+# @doc """
+#     blockdim(t::AbstractTensorMap, c::Sector) -> Base.Dims
 
-Return the dimensions of the block of a tensor corresponding to a coupled sector `c`.
-""" blockdim(::AbstractTensorMap, ::Sector)
+# Return the dimensions of the block of a tensor corresponding to a coupled sector `c`.
+# """ blockdim(::AbstractTensorMap, ::Sector)
 
-@doc """
+"""
     fusiontrees(t::AbstractTensorMap)
 
 Return an iterator over all splitting - fusion tree pairs of a tensor.
-""" fusiontrees(::AbstractTensorMap)
+"""
+fusiontrees(t::AbstractTensorMap) = tensorstructure(t).fusiontreelist
+
+# auxiliary function
+@inline function trivial_fusiontree(t::AbstractTensorMap)
+    sectortype(t) === Trivial ||
+        throw(SectorMismatch("Only valid for tensors with trivial symmetry"))
+    spaces1 = codomain(t).spaces
+    spaces2 = domain(t).spaces
+    f₁ = FusionTree{Trivial}(map(x -> Trivial(), spaces1), Trivial(), map(isdual, spaces1))
+    f₂ = FusionTree{Trivial}(map(x -> Trivial(), spaces2), Trivial(), map(isdual, spaces2))
+    return (f₁, f₂)
+end
```

Comment:
just out of curiosity, do you know if `only(fusiontrees(t))` gives the same result? Is there a noticable speedup?

Reply:
I was indeed doubting about this. Calling `fusiontrees(t)` requires a cache lookup; whereas this is probably almost completely compiled away.